endpoint_info.go

```diff
@@ -159,7 +159,11 @@ func (ep *endpoint) Info() EndpointInfo {
 		return ep
 	}
 
-	return sb.getEndpoint(ep.ID())
+	if epi := sb.getEndpoint(ep.ID()); epi != nil {
+		return epi
+	}
```

Comment:
Can you please explain a bit more about this fix ?
`getEndpoint` returns nil if the endpoint id doesnt match. So, what is the purpose of this fix and how does it impact docker/docker#18009 ?